### DISCUSSION TITLE
shortcut for literals in objects?

### DISCUSSION THREAD

**Discussion by rickmed** [colinhacks/zod#2043] [Q&A] [answered]
To write 
```
const Schema = z.object({
    name: "mia"
})
```

instead of 
```
const Schema = z.object({
    name: z.literal("mia")
})
```

**Comment by JacobWeisenburger** [comment]
Can you explain a little more about your use case?
What is the purpose of this?

**Comment by rickmed** [comment]
I'm trying to use zod with unit testing. Something like:
```javascript
function assertThing(receive, userId, actions) {

	const schema = z.object({
		error: z.string().regex(/Error/),
		userId: z.literal(userId),
		actions: z.array(),  //?
	}).strict()

	return schema.safeParse(rec)
}

const receive = {
	error: "someErrr",
	userId: "nwn92fu",
	actions: ["hi", "hello"],
}

const res = assertThing(receive, "nwn92f", 60)
```
 But I'm finding out that maybe is not the right use case (?)

**Comment by JacobWeisenburger** [comment]
```ts
return schema.safeParse( rec )
// Cannot find name 'rec'
```

**Comment by rickmed** [comment]
They code above is pseudo-code. It wouldn't work anyway because in a unit test I would want, for example, to validate that the .actions field is equal to ["hi, "hello"], and I'm not sure there's an easy way to do it with zod.

**Comment by JacobWeisenburger** [comment]
is `actions` supposed to be an array of literals? or an [enum](https://github.com/colinhacks/zod#zod-enums)?

**Comment by rickmed** [comment]
array of literals

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
const schema = z.object( {
    actions: z.tuple( [
        z.literal( 'hi' ),
        z.literal( 'hello' ),
    ] ),
} )
console.log( schema.parse( { actions: [ 'hi', 'hello' ] } ) )
// { actions: [ 'hi', 'hello' ] }
```

or perhaps this?

```ts
const schema = z.object( {
    actions: z.union( [
        z.literal( 'hi' ),
        z.literal( 'hello' ),
    ] ).array(),
} )
console.log( schema.parse( { actions: [ 'hi', 'hello', 'hello', 'hi' ] } ) )
// { actions: [ 'hi', 'hello', 'hello', 'hi' ] }
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger
